mlflow/gateway/providers/anthropic.py

```diff
@@ -110,6 +227,77 @@ def __init__(self, config: RouteConfig) -> None:
         }
         self.base_url = "https://api.anthropic.com/v1/"
 
+    async def chat_stream(
+        self, payload: chat.RequestPayload
+    ) -> AsyncIterable[chat.StreamResponsePayload]:
+        payload = jsonable_encoder(payload, exclude_none=True)
+        self.check_for_model_field(payload)
+        stream = send_stream_request(
+            headers=self.headers,
+            base_url=self.base_url,
+            path="messages",
+            payload={
+                "model": self.config.model.name,
+                **AnthropicAdapter.chat_streaming_to_model(payload, self.config),
+            },
+        )
+
+        indices = []
+        metadata = {}
+        async for chunk in stream:
+            chunk = chunk.strip()
+            if not chunk:
+                continue
+
+            # No handling on "event" lines
+            prefix, content = chunk.decode("utf-8").split(":", 1)
+            if prefix != "data":
+                continue
+
+            # See https://docs.anthropic.com/claude/reference/messages-streaming
+            resp = json.loads(content)
+            if resp["type"] not in (
+                "message_start",
+                "message_delta",
+                "content_block_start",
+                "content_block_delta",
+            ):
+                continue
+
+            # response id and model are only present in `message_start`
+            if resp["type"] == "message_start":
+                metadata["id"] = resp["message"]["id"]
+                metadata["model"] = resp["message"]["model"]
+                continue
+
+            index = resp.get("index")
+            if index is not None and index not in indices:
+                indices.append(index)
+
+            resp.update(metadata)
+            if resp["type"] == "message_delta":
+                for index in indices:
+                    yield AnthropicAdapter.model_to_chat_streaming(
+                        {**resp, "index": index},
+                        self.config,
+                    )
```

Comment:
`stop_reason` is in `message_delta` but `index` is in `content_block_*`. So we should emit a stop event with `stop_reason` for each `index` we have seen before